spec/graphql/query/executor_spec.rb

```diff
@@ -168,6 +168,21 @@
       end
     end
 
+    describe 'if a non-nil value coerces to nil for a non-null field' do
+      let(:query_string) {%| query noMilk { cow { name cantBeNullButIsOnCoersion } }|}
+      it 'turns into error message and nulls the entire selection' do
+        expected = {
+          "data" => { "cow" => nil },
+          "errors" => [
+            {
+              "message" => "Cannot return null for non-nullable field cantBeNullButIsOnCoersion"
```

Comment:
This isn't really a helpful error for either the client or the server developers.  I would prefer if we reported these errors as internal errors.